controllers/nodefeaturediscovery_controller.go

```diff
@@ -169,6 +173,50 @@ func (r *NodeFeatureDiscoveryReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	// If the resources are to be deleted, first check to see if the
+	// deletion timestamp has a value.
+	if instance.DeletionTimestamp != nil {
+
+		// Attempt to delete all components. If it fails, return
+		// a warning letting users know the deletion failed, and
+		// then call the reconciler once more to see if the error
+		// can be corrected.
+		if err := r.deleteComponents(ctx); err != nil {
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "Deletion failed", "Failed to delete components: %v", err)
+			return reconcile.Result{}, err
+		}
+		r.Recorder.Eventf(instance, corev1.EventTypeNormal, "Deletion succeeded", "Succeeded to delete all components")
+
+		// Check if all components are deleted. If they're not,
+		// then call the reconciler but wait 10 seconds before
+		// checking.
+		if r.doComponentsExist(ctx) {
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		}
+
+		// If all components are deleted, then remove the finalizer
+		if hasFinalizer(instance, finalizer) {
+			removeFinalizer(instance, finalizer)
+			if err := r.Update(ctx, instance); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// Add the defined finalizer as a finalizer to the instance if it does not exist
```

Comment:
```suggestion
	// add finalizer
```
clean and simple comment